Framework/Kernel/inc/MantidKernel/IPropertyManager.h

```diff
@@ -467,6 +476,24 @@ class MANTID_KERNEL_DLL IPropertyManager {
     }
     return this;
   }
+
+  template <typename T>
+  IPropertyManager *setTypedProperty(const std::string &name,
+                                     const std::vector<T> &value,
+                                     const boost::true_type &) {
+    std::vector<boost::shared_ptr<DataItem>> tmp(value.size());
+
+    for (size_t i = 0; i < value.size(); i++)
+      tmp[i] = boost::static_pointer_cast<DataItem>(value[i]);
```

Comment:
The boost::true_type is only allowed where T is `boost::is_convertible` to DataItem. The false type defaults to the `setTypedProperty` method which will fail when attempting the dynamic_cast.